reinforcement_learning/README.md

```diff
@@ -0,0 +1,93 @@
+
+Before training a dialogue policy, you need to prepare a configuration file.
+A default configuration for each supported algorithm is provided in the `config/rl` folder.
+
+A dialogue policy is trained by executing the following command:
+
+```bash
+python -m reinforcement_learning.train_dialogue_policy -c <training_configuration>
+```
+
+The training logs are stored in Weight and Biases.
+
+### Training Configuration
+
+The training configuration is used to specify paths to participants configuration files, model's parameters, training hyperparameters, and other options related to training. Below is an empty configuration file with main options:
+
+```yaml
+# Participants' configuration
+usersim_config: <path to user simulator configuration>
+agent_config: <path to IAI MovieBot configuration>
+
+# Model's parameters
+model:
+  algorithm: <dqn OR a2c>
+  policy_input_size: 
+  hidden_size: 
+  model_path: <path to saved model>
+
+# Training hyperparameters
+hyperparams:
+  hyperparam_name: hyperparam_value
+  ...
+
+use_intents: 
+turn_penalty: 
+no_nlu: 
+
+debug: False
+```
+
+**Participants' configuration**: The participants' configuration files are used to specify the parameters of the user simulator and the agent.
+
+**Model's parameters**: The model's parameters are used to specify the parameters of the dialogue policy model. The `algorithm` parameter is used to specify the type of the algorithm used, which can be either `dqn` or `a2c`. The `policy_input_size` parameter is used to specify the size of the input of the policy network. The `hidden_size` parameter is used to specify the size of the hidden layers of the policy network. The `model_path` parameter is used to specify the path to the saved model.
+
+**Hyperparameters**: This section is reserved for training hyperparameters such as learning_rate, discount factor, etc. The hyperparameters are algorithm-specific and can be loaded easily in a dictionary.
+
+**use_intents**: Specifies whether the markovian state should include intents.
+
+**turn_penalty**: Specifies the turn penalty value that is included in the reward computation.
+
+**no_nlu**: Specifies whether the user simulator's responses are processed by the agent's NLU module or not. If not, the agent takes the structured representation of the user simulator's responses as input.
+
+## User Simulator
+
+The user simulator is built on top of [UserSimCRS](https://github.com/iai-group/UserSimCRS/tree/main).
+It is an agenda-based user simulator adapted for reinforcement learning.
+In our case, the user simulator needs to send its response directly to the agent instead of letting the dialogue connector handle it.
+
+We note that its architecture is the same as the one used in the UserSimCRS.
+
+## Try dialogue policy
+
+A dialogue policy can be tested with a human user via the terminal. Instead of interacting with the user simulator, the dialogue policy will interact with the user.
```

Comment:
How to configure this? How to run it?

Reply:
For now, we do not use a configuration file but it may change.